content/automate/ManageIQ/Cloud/VM/Provisioning/StateMachines/VMProvision_VM.class/__methods__/retry_placement.rb

```diff
@@ -0,0 +1,29 @@
+#
+# Description: This method updates the provisioning status and retries
+# from the Placement step.
+# Required inputs: status
+#
+
+prov = $evm.root['miq_provision']
+unless prov
+  $evm.log(:error, "miq_provision object not provided")
+  exit(MIQ_STOP)
+end
+status = $evm.inputs['status']
```

Comment:
@mkanoor Is the exception that resulted in the error state exposed to automate so that the script can make a decision based on it, or do I have to determine another way? Additionally, is there a way to have this automatically respect `max_retries`, or am I going to have to implement my own mechanism?